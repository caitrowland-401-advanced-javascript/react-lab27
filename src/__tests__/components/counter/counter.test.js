
import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Counter from '../../../components/counter/counter'

describe('<Counter/>', () => {
  it('is alive at application start', () => {
    let app = shallow(<Counter />);
    expect(app.find('span').exists()).toBeTruthy();
  });
  describe('changes on click', () => {
    it('changes state on click + ', () => {
      let app = mount(<Counter />);
      let button = app.find('#upClick');
      button.simulate('click');
      expect(app.find('#currentCount').text()).toContain(1);
    });

    it('changes state on click -', () => {
      let app = mount(<Counter />);
      let button = app.find('#downClick');
      button.simulate('click');
      expect(app.find('#currentCount').text()).toContain(-1);

    })
})
  it('renders correctly', () => {
    const count = renderer.create(<Counter />).toJSON();
    expect(count).toMatchSnapshot();
  });
});