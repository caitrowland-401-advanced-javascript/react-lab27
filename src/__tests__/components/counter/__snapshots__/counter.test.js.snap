// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter/> renders correctly 1`] = `
<section
  className="counter"
>
  <span
    className="down clicker"
    id="downClick"
    onClick={[Function]}
  >
     - 
  </span>
  <span
    className="count "
    id="currentCount"
  >
    0
  </span>
  <span
    className="up clicker"
    id="upClick"
    onClick={[Function]}
  >
     + 
  </span>
</section>
`;
